### PR TITLE
Week 6 keelan pattersoncaesar

### PR DESCRIPTION
**Pull request by keelanpattersoncaesar**

### CODE REVIEW

File: code-challenges.test.js [hunk 1]
```diff
@@ -19,33 +27,239 @@ const people = [
...
+  ]
+  it("takes in an array of objects and returns an array with a sentence about each person with their name capitalized", () => {
+
+    expect(jobPlusName(people)).toEqual(["Ford Prefect is a hitchhiker.", "Zaphod Beeblebrox is president of the galaxy.", "Arthur Dent is a radio employee."])
+  })
+})
+
+// ReferenceError: jobPlusName is not defined, good fail
+
+// Expected output: ["Ford Prefect is a hitchhiker.", "Zaphod Beeblebrox is president of the galaxy.", "Arthur Dent is a radio employee."]
```
Comment (SunkissedQueen) [inline code-challenges.test.js:50]: Proud of the work you put into this challenge

File: code-challenges.test.js [hunk 2]
```diff
@@ -19,33 +27,239 @@ const people = [
...
+// (It iterates through the rest of the array...)
+
+//       [
+//         'Ford Prefect is a hitchhiker.',
+//         'Zaphod Beeblebrox is president of the galaxy.',
+//         'Arthur Dent is a radio employee.'
+//       ]
+
+// WHAT I LEARNED: Console.log() is your friend. I used it on almost every line (I deleted them all to make it more readable) so I could track exactly what was happening and where. This problem took me the longest to figure out so, I left it for last. 
+
```
Comment (SunkissedQueen) [inline code-challenges.test.js:109]: It certainly does help. I know it helps to see what the variables are producing. Great observation

File: code-challenges.test.js [hunk 3]
```diff
@@ -19,33 +27,239 @@ const people = [
...
+  const hodgepodge1 = [23, "Heyyyy!", 45, -10, 0, "Yo", false];
+  const hodgepodge2 = [5, "Hola", 43, -34, "greetings", true];
+
+  it(" takes in a mixed data array and returns an array of only the REMAINDERS of the numbers when divided by 3", () => {
+    expect(numRem(hodgepodge1)).toEqual([2, 0, -1, 0]);
+    expect(numRem(hodgepodge2)).toEqual([2, 1, -1]);
+  });
+});
+
+// ReferenceError: remainingRemainders is not defined; good fail!
```
Comment (SunkissedQueen) [inline code-challenges.test.js:131]: woohoo

File: code-challenges.test.js [hunk 4]
```diff
@@ -19,33 +27,239 @@ const people = [
...
+
+// RESULT:
+// console.log
+// [ 2, 0, -1, 0 ]
+// at numRem (code-challenges.test.js:93:11)
+
+// [ 2, 1, -1 ]
+// at numRem (code-challenges.test.js:93:11)
+
+// What I learned: .filter() creates a new array, and .map() creates a new array. They do not change the original array, which makes iteration easier (and it hits all index positions too!).
```
Comment (SunkissedQueen) [inline code-challenges.test.js:193]: Loving the notes

File: code-challenges.test.js [hunk 5]
```diff
@@ -19,33 +27,239 @@ const people = [
...
+  const cubeAndSum1 = [2, 3, 4];
+  const cubeAndSum2 = [0, 5, 10];
+
+  it("takes in an array of numbers and returns the sum of all the numbers cubed", () => {
+    expect(cubeSumer(cubeAndSum1)).toEqual(99);
+    expect(cubeSumer(cubeAndSum2)).toEqual(1125);
+  });
+});
+
+// ReferenceError: cubeSummer is not defined; good fail :)
```
Comment (SunkissedQueen) [inline code-challenges.test.js:214]: good work on these test structures

File: code-challenges.test.js [hunk 6]
```diff
@@ -19,33 +27,239 @@ const people = [
...
+// RESULT:
+// console.log
+
+// 99
+// at cubeSumer (code-challenges.test.js:145:11)
+
+// 1125
+// at cubeSumer (code-challenges.test.js:145:11)
+
+// What I learned: You can chain multiple arrow functions together. Originally, I planned to separate the .map() and .reduce() into separate functions. Also, console.log() is amazing because it allows you to see what is happening and where. I also learned to STOP using return prematurely. That has been tripping me up for a while now.
```
Comment (SunkissedQueen) [inline code-challenges.test.js:265]: Great work. Please place whether or not your functions passed

File: interview-questions.md [hunk 7]
```diff
@@ -7,44 +7,55 @@ Then, research the question to expand on your answer. Even if you feel you have
 
 1. As a developer, I am creating a Rails application with a model called Cohort that has_many students, but OOPS! I forgot to add the foreign key. How can I fix this mistake? What is the name of the foreign key? Would the foreign key be on the Cohort model or the Student model?
 
-  Your answer:
+  Your answer: Migrations can be used to update the database! I need to research the rest. 
 
-  Researched answer:
+  Researched answer: A developer can perform a migration to create a relationship between two databases using a foreign key. The foreign key would be named has_many :students and I think it would be on the Cohort model.
```
Comment (SunkissedQueen) [inline interview-questions.md:12]: Please review this section of the syllabus 
https://github.com/learn-academy-2022-alpha/Syllabus/blob/main/rails/associations.md

File: interview-questions.md [hunk 8]
```diff
@@ -7,44 +7,55 @@ Then, research the question to expand on your answer. Even if you feel you have
...
 
 
 
 2. Which RESTful routes must always be passed params? Why?
 
-  Your answer:
+  Your answer: I'm guessing it would have to be SHOW, EDIT, UPDATE, and DESTROY.
 
-  Researched answer:
+  Researched answer: I was correct because each of those RESTful routes need to be directly identified in order to have an action be taken on them. How can you delete something if you don't know what you're deleting?
```
Comment (SunkissedQueen) [inline interview-questions.md:20]: True

File: interview-questions.md [hunk 9]
```diff
@@ -7,44 +7,55 @@ Then, research the question to expand on your answer. Even if you feel you have
...
 
-  Your answer:
+  Your answer: 'rails generate controller' creates a controller for your rails application, 'rails generate model' creates a model, and 'rails generate migration' updates your database!
 
-  Researched answer:
+  Researched answer: Like I stated, 'rails generate controller' creates a controller for your rails application. The command 'rails generate model' creates a db model which means we are creating the shape of our database (what columns, rows), Lastly, 'rails generate migration' allows you to add new columns to your db after it has already been created.
 
 
 
 4. Consider the Rails routes below. What is the name of the controller method that would be called by each route? What action would each of the controller methods perform?
```
Comment (SunkissedQueen) [inline interview-questions.md:33]: I can see the confidence

File: interview-questions.md [hunk 10]
```diff
@@ -7,44 +7,55 @@ Then, research the question to expand on your answer. Even if you feel you have
...
 action: "PATCH"  location: /students/2      
-
+UPDATE; Pushes updates made on a student in the database
 action: "DELETE" location: /students/2      
-
+DESTROY; Deletes a student!
 
 
 5. As a developer, you are making an application to manage your to do list. Create 10 user stories that will help you get your application started. Read more about [user stories](https://www.atlassian.com/agile/project-management/user-stories).
+
```
Comment (SunkissedQueen) [inline interview-questions.md:51]: good

File: rails-commenting.rb [hunk 11]
```diff
@@ -6,25 +6,25 @@
 
 # FILE: app/controller/blog_posts_controller.rb
 
-# ---1)
+# ---1) // This is the controller for the blog posts.
```
Comment (SunkissedQueen) [inline rails-commenting.rb:9]: what do you see about this relationship

File: rails-commenting.rb [hunk 12]
```diff
@@ -6,25 +6,25 @@
...
   end
 
-  # ---3)
+  # ---3) // This is the method that displays the blog post. It uses the find method to find the blog post by its unique ID from the postgreSQL database. The ID is passed in as a parameter.
   def show
     @post = BlogPost.find(params[:id])
   end
 
-  # ---4)
+  # ---4) // Creates a new blog post.
```
Comment (SunkissedQueen) [inline rails-commenting.rb:21]: What is unique about this method?

**Review by SunkissedQueen — CHANGES_REQUESTED** [review]
High five Keelan! You have knocked it out the park with you efforts. Proud of all you put into all you assessments. Please review the notes!

File: code-challenges.test.js (hunk 6 above)
Comment (keelanpattersoncaesar) [reply code-challenges.test.js:265]: I mentioned it on Line 214

File: rails-commenting.rb (hunk 11 above)
Comment (keelanpattersoncaesar) [reply rails-commenting.rb:9]: There is a parent-child relationship, with BlogPostController being the child

File: rails-commenting.rb (hunk 12 above)
Comment (keelanpattersoncaesar) [reply rails-commenting.rb:21]: It creates a form where you can submit the data. The create method on line 26 will push it to the database.

**Comment by keelanpattersoncaesar** [comment]
I'm not sure if you wanted me to comment on this commit or re-push what I have worked on. The changes seems pretty small, so I just commented on this PR with a couple of elaborated thoughts. Please let me know if I should take another action.

File: code-challenges.test.js (hunk 6 above)
Comment (SunkissedQueen) [reply code-challenges.test.js:265]: I see that good failure for the test which is needed. We also expect to see when the function passes, which occurs after you create it.

**Review by SunkissedQueen — APPROVED** [review]
Keelan, I appreciate the work you give everything. Links were left for you to look over for questions that the concept was looking shaky. I am approving this because i know you will do the research.

**Comment by keelanpattersoncaesar** [comment]
Thank you for all inputs. I really appreciate it!

**Pull request merged**